Sandbox/conan-recipe/conanfile.py

```diff
@@ -0,0 +1,159 @@
+import json
+import os
+import re
+from conans import ConanFile, CMake, tools
+
+required_conan_version = ">=1.33.0"
+
+class NmosCppConan(ConanFile):
+    name = "nmos-cpp"
+    description = "An NMOS C++ Implementation"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/garethsb/nmos-cpp"
+    topics = ("NMOS")
+
+    settings = "os", "compiler", "build_type", "arch"
+    # no "shared" option support
+    # is anything else required to support "fPIC"?
+    options = {"fPIC": [True, False]}
+    default_options = {"fPIC": True}
+
+    exports_sources = ["CMakeLists.txt"]
+    # is this the right choice of generators?
+    generators = "cmake", "cmake_find_package"
```

Comment:
Since your CMake prefers the CONFIG packages

```suggestion
    generators = "cmake", "cmake_find_package_multi"
```

Reply:
Makes sense. I was sort of surprised that I needed the `cmake` generator as well, but it all goes wrong without?

Reply:
Hmm, the CMake only sets `CMAKE_FIND_PACKAGE_PREFER_CONFIG` when running the `cmake_find_package_multi` generator `if(CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)` (i.e. CMake's Visual Studio generators). It uses the `cmake_find_package` generator otherwise.
Does that make a difference to the answer?

I'm not sure I've seen in the Conan documentation which CMake generators it uses for different platforms? And I'm assuming it never sets `CMAKE_CONFIGURATION_TYPES` and always `CMAKE_BUILD_TYPE` even on Windows and builds Debug and Release packages separately?

Reply:
Ah, after thinking about this a bit more, and finding https://github.com/conan-io/conan/issues/8538#issuecomment-805670963, I went with switching to `"cmake", "cmake_find_package_multi"` as you suggested but also adding
```py
self._cmake.definitions["CMAKE_FIND_PACKAGE_PREFER_CONFIG"] = True
```